### DISCUSSION TITLE
Self referencing includes?

### DISCUSSION THREAD

**Discussion by webnoob** [prisma/prisma#2960] [Q&A] [answered]
Hey,

In SQL, I would do something like:
```sql
select
*
from 
MessgageThread as mt
inner join Arc as a on mt.arcId = a.id
inner join User as u on mt.recipientUserId = u.id
left join ArcClaim as ac on ac.userId = u.id and ac.arcId = a.id
```
to get the claim associated with a user AND arc.

I'd like to replicate this in prisma but I don't know how / not sure it's possible.
```js
const models = this.photonService.messageThread.findMany({
  includes: {
    recipientUser: {
      include: {
        arcClaims: {
          where: {
            arcId: /*USE MessageThread.Arc.Id HERE*/
          }
        }
      }
    },
    arc: {
      select: {
        name: true
      }
    }
  }
})
```
Is it possible to self reference `/*USE MessageThread.Arc.Id HERE*/` somehow so I can avoid extra queries?

I've not posted my schema as I believe the SQL query explains things well enough so I hope there is enough info :)

**Comment by ryands17** [comment, marked as answer]
Hey @webnoob :wave: 
Unfortunately this is not possible directly so as a workarouind you would need to use a raw query via `prisma.queryRaw`. Could you open a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) explaining the above use case?

**Reply by webnoob** [reply]
I thought this might be the case. Yes of course, I'll add it. I've found myself needing it a few times now so finally decided to ask as I didn't think it was possible.

Thanks for the quick response.

**Reply by webnoob** [reply]
FR added: https://github.com/prisma/prisma/issues/2961

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
